confdgnmi/testtool/OpenConfig/01__interfaces.robot

```diff
@@ -0,0 +1,195 @@
+*** Settings ***
+Documentation    Core test suite for OpenConfig telemetry/gNMI tests,
+...              the ``openconfig-interfaces.yang`` model.
+...
+...              Test cases were created with expectation of YANG model ``revision "2022-10-25"``.
+Test Tags        OpenConfig  interfaces
+
+Test Teardown   Clear Interfaces test state
+
+Resource         openconfig.resource
+Resource         interfaces.resource
+
+Resource         ../gNMI_Interface/gNMIClient.resource
+Suite Setup      Setup gNMI Client
+Suite Teardown   Close gNMI Client
+
+
+*** Keywords ***
+Clear Interfaces test state
+    Cleanup Last request Results
+    Cleanup GetRequest Parameters
+
+
+*** Variables ***
+${INTERFACE}     MgmtEth0/RP0/CPU0/0    # TODO - move to YAML
+
+
+*** Test Cases ***
+Supported models include "openconfig-interfaces"
+    [Documentation]    Verify that ``openconfig-interfaces`` is among the models
+    ...                advertised as supported in the ``CapabilityResponse``.
+    [Tags]  sanity
+    Supported Models Should Include  openconfig-interfaces
+
+Get "interfaces" with no parameters
+    [Documentation]    Verify that basic ``GetRequest`` can be issued with ``path=interfaces``
+    ...                parameter and OK response is received (ignoring data contents/payload).
+    [Tags]  sanity
+    Verify Get of  interfaces
+
+Get "interfaces" with "type" parameter
+    [Documentation]    Verify that OK response is retrieved for ``GetRequest`` having ``type=``
+    ...                parameter with any of the defined ``DataType``s.
+    [Tags]  type
+    [Template]  Verify path "${path}" with DataType ${type}
+    interfaces  ALL
+    interfaces  CONFIG
+    interfaces  STATE
+    interfaces  OPERATIONAL
+
+Get "interfaces" for various "path" parameter values
+    [Documentation]    Verify that various valid formats of root container/list
+    ...                can be requested using ``path=`` parameter, and are responded to correctly.
+    [Tags]  path  costly
+    [Template]    Verify Get of
+    # no namespace
+    /interfaces
+    interfaces/
+    /interfaces/
+    interfaces/interface
+    /interfaces/interface
+    /interfaces/interface/
+    # root namespace
+    openconfig-interfaces:interfaces
+    /openconfig-interfaces:interfaces
+    openconfig-interfaces:interfaces/
+    /openconfig-interfaces:interfaces/
+    # non-root namespace
+    interfaces/openconfig-interfaces:interface
+    interfaces/openconfig-interfaces:interface/
+    /interfaces/openconfig-interfaces:interface
+    /interfaces/openconfig-interfaces:interface/
+    # both namespaces
+    openconfig-interfaces:interfaces/openconfig-interfaces:interface
+    /openconfig-interfaces:interfaces/openconfig-interfaces:interface
+    openconfig-interfaces:interfaces/openconfig-interfaces:interface/
+    /openconfig-interfaces:interfaces/openconfig-interfaces:interface/
```

Comment:
legit argument, am not familiar with standard in detail, i assumed that both with/without namespace path must be supported for any path even when path is not needed due to conflicts...

Reply:
I agree that slashes are redundant due to client parsing, and I will remove them.
Other path cases are still useful IMHO - and can be useful indication of device's gNMI capabilities/implementation details. - It can show whether device can/not use namespaces in paths etc.